triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Converter.scala

```diff
@@ -322,6 +331,109 @@ final class Converter(
       "config" -> config,
     )
 
+  def fromCommand(command: Command): SValue = command match {
+    case Command.Create(templateId, argument) =>
+      record(
+        createCommandTy,
+        "templateArg" -> fromAnyTemplate(
+          TTyCon(templateId),
+          argument,
+        ),
+      )
+
+    case Command.ExerciseTemplate(_, contractId, _, choiceArg) =>
+      record(
+        exerciseCommandTy,
+        "contractId" -> contractId,
+        "choiceArg" -> choiceArg,
+      )
+
+    case Command.ExerciseInterface(_, contractId, _, choiceArg) =>
+      record(
+        exerciseCommandTy,
+        "contractId" -> contractId,
+        "choiceArg" -> choiceArg,
+      )
+
+    case Command.CreateAndExercise(templateId, createArg, _, choiceArg) =>
+      record(
+        createAndExerciseCommandTy,
+        "templateArg" -> fromAnyTemplate(TTyCon(templateId), createArg),
+        "choiceArg" -> choiceArg,
+      )
+
+    case Command.ExerciseByKey(templateId, contractKey, _, choiceArg) =>
+      record(
+        exerciseByKeyCommandTy,
+        "tplTypeRep" -> SValue.STypeRep(TTyCon(templateId)),
+        "contractKey" -> contractKey,
+        "choiceArg" -> choiceArg,
+      )
+
+    case _ =>
+      throw new ConverterException(
+        s"${command.getClass.getSimpleName} is an unexpected command type"
+      )
+  }
+
+  def fromCommands(commands: Seq[Command]): SValue = {
+    SList(commands.map(fromCommand).to(FrontStack))
```

Comment:
If `fromCommand` returned an either, we'd have to work a lot harder here (to also return an either), or convert via `orConverterException`